DSpace/dspace-angular
# 401 login page and 403 page

## References
Fixes #923 

## Description
This PR changes the `redirectOn404Or401` (now called `redirectOn4xx`) operator to redirect the user to the login page when receiving a 401 and the user isn't logged in. It now also supports a redirect to a 403 page.

## Instructions for Reviewers
How to test:
- Visit a restricted dso while logged out. Visiting the page should redirect you to the login page and after logging in, you should end up on the requested page.
- Visit a forbidden dso while logged in as a user that doesn't have any rights to view this dso. You should be redirected to a 403 page.

## Checklist
_This checklist provides a reminder of what we are going to look for when reviewing your PR. You need not complete this checklist prior to creating your PR (draft PRs are always welcome). If you are unsure about an item in the checklist, don't hesitate to ask. We're here to help!_
